### PR TITLE
New ingest workflow

### LINKED ISSUE #76: Stop triggering rebuild on edits to annotations
A rebuild is not needed every single time, and manual rebuilds are easy to re-trigger. This will result in about the same amount of effort (triggering 50% of the time vs. the status quo of cancelling 50% of the time), but is more logical and results in fewer Slack notifications. 

(See [this Slack thread](https://bedfordlab.slack.com/archives/CTZKJC7PZ/p1594147551229600) for the original context.)

### PR DESCRIPTION
**Pull request by joverlee521**
Resolves #76 

- Add `--fetch` option to the `ingest-gisaid` script. This determines whether the ingest script runs by fetching new sequences from GISAID or just use the existing `gisaid.ndjson` file on AWS. 
- Create separate workflows for the build shepherds to have more control of when to fetch new sequences from GISAID.

### CODE REVIEW

**Comment by emmahodcroft** [comment]
This is looking great Jover, thank you!

File: bin/ingest-gisaid [hunk 1]
```diff
@@ -37,15 +37,29 @@ esac
...
+fetch=0
+
+for arg; do
+    case "$arg" in
+        --fetch)
+            fetch=1;;
+        *)
+    esac
+done
+
```
Comment (kairstenfay) [inline bin/ingest-gisaid:49]: Taking a look at this PR now, but I have a small request: now that we have arguments, could we add a `--help` function?

Comment (kairstenfay) [reply bin/ingest-gisaid:49]: e.g. [this one in Seattleflu codebase](https://github.com/seattleflu/backoffice/blob/85be1faa2835ef7fc5b2848cd914b7bd4c229baa/bin/generate-and-upload-uw-retro-redcap-dets#L38)

**Comment by kairstenfay** [comment]
I'm looking through your commit messages
> When working on a feature branch, we would always want to fetch the
latest GISAID sequences.

Why is this the case? The answer is not apparent to me.

File: README.md [hunk 2]
```diff
@@ -11,15 +11,20 @@ If you're using Pipenv (see below), then run commands from `./bin/…` inside a
...
 
 AWS credentials are stored in this repository's secrets and are associated with the `nextstrain-ncov-ingest-uploader` IAM user in the Bedford Lab AWS account, which is locked down to reading and publishing only the `gisaid.ndjson`, `metadata.tsv`, and `sequences.fasta` files and their zipped equivalents in the `nextstrain-ncov-private` S3 bucket.
 
 ## Manually triggering the automation
-You can manually trigger the full automation by running `./bin/trigger ingest --user <your-github-username>`.
-If you want to only trigger a rebuild of [nextstrain/ncov](https://github.com/nextstrain/ncov) without re-ingesting data from GISAID first, run `./bin/trigger rebuild --user <your-github-username>`.
-See the output of `./bin/trigger ingest` or `./bin/trigger rebuild` for more information about authentication with GitHub.
+A full run is a now done in 3 steps via manual triggers:
+1. Fetch new sequences and ingest them by running `./bin/trigger fetch-and-ingest --user <your-github-username>`.
+2. Add manual annotations and update location hiearchy as needed and run ingest without fetching new sequences.
```
Comment (kairstenfay) [inline README.md:22]: copy edit: I'd rewrite this slightly as 
> Add manual annotations, update location hiearchy as needed, and run ingest without fetching new sequences.

Comment (kairstenfay) [reply README.md:22]: oops! caught a typo just now in `hierarchy`

**Comment by joverlee521** [comment]
> I'm looking through your commit messages
> 
> > When working on a feature branch, we would always want to fetch the
> > latest GISAID sequences.
> 
> Why is this the case? The answer is not apparent to me.

Ah, this comment was mostly regarding the first time you push a feature branch, the `s3://nextstrain-ncov-private/branch/<branch-name>/gisaid.ndjson.gz` object would not exist on AWS. So the workflow must fetch the sequences from GISAID.

**Comment by kairstenfay** [comment]
> > I'm looking through your commit messages
> > > When working on a feature branch, we would always want to fetch the
> > > latest GISAID sequences.
> > 
> > 
> > Why is this the case? The answer is not apparent to me.
> 
> Ah, this comment was mostly regarding the first time you push a feature branch, the `s3://nextstrain-ncov-private/branch/<branch-name>/gisaid.ndjson.gz` object would not exist on AWS. So the workflow must fetch the sequences from GISAID.

Oh, that makes sense. Thanks!

**Comment by emmahodcroft** [comment]
I don't know how ready this is looking, but if you guys think it's ready to go, it would be great to get this implemented (with appropriate heads-up and notes on the new commands to all the shepherds!). We've been working on cleaning up the `parse_metadata.py` script and often this would be assisted by being able to push to `gisaid_annotations.tsv` and get new `metadata.tsv` files, but at the moment this screws up whoever's building next, so has been a little hobbled. 
However, no rush if it's not ready!

**Pull request merged**